### PR TITLE
Add Snowflake target and simplify

### PR DESCRIPTION
**Pull request by ian-r-rose**
This adds Snowflake as an option for this template. It also removes a few options from the copier flow:

* No longer have `use_dbt` as an option, this always creates a dbt project.
* Remove `use_orchestrator` as an option, we can bring it back if we need it.

With this PR we're in better shape to quickly set up some dbt projects for CalHR

### CODE REVIEW

File: ci/test.sh [hunk 1]
```diff
@@ -6,30 +6,33 @@ TEAM_NAME="Test team"
...
-# Run quality checks
-pre-commit run --all-files
-
-# Verify that the docs build
-dbt deps --project-dir=transform
-dbt docs generate --project-dir=transform
-cp -r transform/target docs/dbt_docs
-mkdocs build
+# Run copier.
+for TARGET in Snowflake BigQuery; do
```
Comment (britt-allen) [inline ci/test.sh:10]: is Snowflake BigQuery missing a comma or is BigQuery intended to be there at all?

**Review by britt-allen — APPROVED** [review]
Looks good, have you tested the changes and new flow? Is this something that can be easily done in a new conda environment? If so, I'd be happy to walk through again. If not, still happy to walk through assuming that's useful but would love to know how to make it less frictiony.

**Review by britt-allen — CHANGES_REQUESTED** [review]
I am actually getting a failure: https://github.com/cagov/caldata-infrastructure-template/actions/runs/5127313142/jobs/9222791711?pr=8

File: ci/test.sh (hunk 1 above)
Comment (ian-r-rose) [reply ci/test.sh:10]: No comma missing, just bash being a weird programming language

**Comment by ian-r-rose** [comment]
> Looks good, have you tested the changes and new flow? Is this something that can be easily done in a new conda environment? If so, I'd be happy to walk through again. If not, still happy to walk through assuming that's useful but would love to know how to make it less frictiony.

Please do give it a spin! The script `test.sh` runs through it in CI, but it's not really a substitute for a human user.


> I am actually getting a failure: https://github.com/cagov/caldata-infrastructure-template/actions/runs/5127313142/jobs/9222791711?pr=8

Should be working now, I had to set up some new authentication stuff in this repo

**Comment by britt-allen** [comment]
I have merged but created a new issue now to remind myself to give it a
spin.

**Comment by ian-r-rose** [comment]
Thanks for the review!

**Pull request merged**